tests/clone.rs

```diff
@@ -0,0 +1,48 @@
+extern crate glib;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use glib::clone;
+
+struct State {
+    started: bool,
+}
+
+impl State {
+    fn new() -> Self {
+        Self {
+            started: false,
+        }
+    }
+}
+
+#[test]
+fn clone_closure() {
+    let state = Rc::new(RefCell::new(State::new()));
+    assert_eq!(state.borrow().started, false);
+
+    let closure = {
+        clone!(@weak state => move || {
```

Comment:
Maybe also add a test for strong, weak+strong mixed and the weak-failure-return-value :)

Reply:
Here are the current tests:
weak only,
weak and strong (strong is by default, i.e. no keyword),
weak failure with return value.
What is missing?

Reply:
Nothing, I was reviewing before you added that last commit with the new tests :)